File: elixir/binary_sequence.ex
defmodule BinarySequenece do
  def run(number) do
    sequences =
      number
      |> number_to_binary("")
      |> String.split(~r/1+/, trim: true)

    if length(sequences) > 1 do
      Enum.map(sequences, fn seq ->
        String.length(seq)
      end)
      |> Enum.max()
    else
      0
    end
  end

  defp number_to_binary(number, result) when number > 0 do
    quotient = div(number, 2)
    reminder = rem(number, 2)
    result = to_string(reminder) <> result
    number_to_binary(quotient, result)
  end

  defp number_to_binary(quotient, result) when quotient == 0, do: result
end

File: javascript/binary_sequence.js
function getLongestSequence(number) {
    let binaryOfNumber = getBinary(number);
    let sequences = binaryOfNumber.replace(/^0+/, "").split(/1/).filter((input) => input != "");
    if(sequences.length > 1) {
      return sequences.reduce((max, seq) => {
        if(seq) {
          return (seq.length > max) ? seq.length : max;
        }
      }, 0);
    } 
    return 0;
}

function getBinary(number) {
    let result = "";
    while(number > 0){
        number = ~~(number / 2);
        let remainder = number % 2;
        result += remainder.toString();
    }
    return result;
}

console.log(getLongestSequence(5));